mongodb/db_test.go

```diff
@@ -16,11 +18,43 @@ func BenchmarkMongoDBRandomReadsWrites(b *testing.B) {
 }
 
 func BenchmarkMongoDBRandomBatchWrites(b *testing.B) {
-	db, err := NewDB("mongodb://localhost:27017", "test", "collection")
+	db, err := NewDB("mongodb://root:example@localhost:27017", "test", "collection")
 	require.Nil(b, err)
 	defer db.Close()
 
 	db.DeleteAll()
 	batch := db.NewBatch()
 	dbtest.BenchmarkRandomBatchWrites(b, db, batch)
-}
\ No newline at end of file
+}
+
+func TestMongoDBSetBson(t *testing.T) {
+
+	db, err := NewDB("mongodb://root:example@localhost:27017", "test", "collection")
+	db.DeleteAll()
+	require.NoError(t, err)
+	key1 := []byte("key-2")
+	var bsonval = bson.D{{"custom", "value-3"}}
+	bsonbytes, _ := bson.Marshal(bsonval)
+
+	db.Set(key1, bsonbytes)
+	value2, err := db.Get(key1)
+	require.NoError(t, err)
+	//require.Equal(t, "value-3", value2)
+	var bsonvalret bson.D
+	err = bson.Unmarshal(value2, &bsonvalret)
+	require.NoError(t, err)
+	require.Equal(t, "value-3", bsonvalret.Map()["custom"])
+	defer db.Close()
```

Comment:
defer should be immediately after db is connected. This is to make sure db is always closed even if there is an error. Even an exception happens the defer function will always run.